lib/main.dart

```diff
@@ -41,23 +42,39 @@ class MyApp extends StatelessWidget {
         ChangeNotifierProvider(create: (_) => ScheduleLayoutSettingProvider()),
         ChangeNotifierProvider(create: (_) => SavedSubjectsProvider()),
       ],
-      child: MaterialApp(
-        title: 'IIUM Schedule',
-        theme: ThemeData(primarySwatch: Colors.purple),
-        darkTheme: ThemeData.dark().copyWith(
-          cupertinoOverrideTheme:
-              const CupertinoThemeData(primaryColor: Colors.purple),
-          textButtonTheme: TextButtonThemeData(
-            style:
-                TextButton.styleFrom(foregroundColor: Colors.purple.shade200),
-          ),
-          outlinedButtonTheme: OutlinedButtonThemeData(
-            style: OutlinedButton.styleFrom(
-                foregroundColor: Colors.purple.shade200),
-          ),
-        ),
-        themeMode: ThemeMode.system,
-        home: const MyBody(),
+      child: DynamicColorBuilder(
+        builder: (ColorScheme? lightColorScheme, ColorScheme? darkColorScheme) {
+          return MaterialApp(
+            title: 'IIUM Schedule',
+            theme: ThemeData(
+              colorScheme: lightColorScheme,
+              useMaterial3: true,
+              fontFamily: 'Inter'
+            ),
+            darkTheme: ThemeData.dark().copyWith(
+              // cupertinoOverrideTheme:
+              //     const CupertinoThemeData(primaryColor: Color(0xFF23682B)),
+              // textButtonTheme: TextButtonThemeData(
+              //   style:
+              //       TextButton.styleFrom(foregroundColor: Colors.purple.shade200),
+              // ),
+              // outlinedButtonTheme: OutlinedButtonThemeData(
+              //   style: OutlinedButton.styleFrom(
+              //       foregroundColor: Colors.purple.shade200),
+              // ),
+              useMaterial3: true,
+              textTheme: ThemeData.dark().textTheme.apply(
+                fontFamily: 'Inter'
+              ),
+              primaryTextTheme: ThemeData.dark().textTheme.apply(
+                fontFamily: 'Inter'
+              ),
+              colorScheme: darkColorScheme
```

Comment:
Oops, missed this one.

```suggestion
              colorScheme: darkColorScheme ?? const ColorScheme.dark()),
```